service/inspectImages.go

```diff
@@ -0,0 +1,271 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/openSUSE/helm-mirror/formatter"
+
+	"github.com/openSUSE/helm-mirror/utils"
+
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ImagesServiceInterface defines a Get service
+type ImagesServiceInterface interface {
+	Images() error
+}
+
+// ImagesService structure definition
+type ImagesService struct {
+	target    string
+	formatter formatter.Formatter
+	logger    utils.Logger
+	io        utils.Io
+	buffer    bytes.Buffer
+}
+
+// NewImagesService return a new instace of ImagesService
+func NewImagesService(target string, formatter formatter.Formatter, logger utils.Logger, io utils.Io) ImagesServiceInterface {
+	return &ImagesService{
+		target:    target,
+		formatter: formatter,
+		logger:    logger,
+		io:        io,
+	}
+}
+
+//Images extracts al the images in the Helm Charts downloaded by the get command
+func (i *ImagesService) Images() error {
+	var shell = ""
+	shell = os.Getenv("SHELL")
+	if shell == "" {
+		i.logger.Info("could not determine shell $SHELL using default `/bin/sh`")
+		shell = "/bin/sh"
+	} else {
+		i.logger.Info("using $SHELL: %s", shell)
+	}
+
+	if !i.io.ExistFile(i.target) {
+		i.logger.Fatal("target %s does not exist", i.target)
+		return errors.New("target " + i.target + " does not exist")
+	}
+
+	var err error
+	if i.io.IsDirectory(i.target) {
+		err = i.processDirectory(shell)
+	} else {
+		err = i.processTgz(i.target, shell)
+	}
+	if err != nil {
+		i.logger.Error("procesing target %s", i.target)
+		return errors.New("procesing target " + i.target)
+	}
+
+	err = i.formatter.Output(i.buffer)
+	if err != nil {
+		i.logger.Error("writing output: %s", err)
+		return err
+	}
+	return nil
```

Comment:
Go std library has native support of `zip` and `tar`. Can you rewrite the code to avoid shelling out? You can take inspiration from [that](http://blog.ralch.com/tutorial/golang-working-with-tar-and-gzip/).